src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -368,6 +368,12 @@ ResetVisitedFields(HTAB *adjacencyList)
 static bool
 AssociateDistributedTransactionWithBackendProc(TransactionNode *transactionNode)
 {
+#ifdef USE_ASSERT_CHECKING
+
+	/* if assertions are disabled, it would give unused variable warning */
+	int32 localGroupId = GetLocalGroupId();
```

Comment:
We usually don't care too much about performance for assert-enabled builds, since Postgres itself gets kind of slow, but ok.

Reply:
actually this was not for optimization but compiler was giving warning when we do not wrap it `USE_ASSERT_CHECKING` when assertions are disabled

Reply:
Marco's remark is saying that you could've left `== GetLocalGroupId()` in the loop since it's within `Assert`

You can also use `PG_USED_FOR_ASSERTS_ONLY` for variables only used in asserts, though in this case you'd rather never call GetLocalGroupId outside assert builds